pkgs/by-name/le/lerc/package.nix

```diff
@@ -23,14 +24,28 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-IHY9QtNYsxPz/ksxRMZGHleT+/bawfTYNVRSTAuYQ7Y=";
   };
 
-  patches = [
-    # https://github.com/Esri/lerc/pull/227
-    (fetchpatch {
-      name = "use-cmake-install-full-dir.patch";
-      url = "https://github.com/Esri/lerc/commit/5462ca7f7dfb38c65e16f5abfd96873af177a0f8.patch";
-      hash = "sha256-qaNR3QwLe0AB6vu1nXOh9KhlPdWM3DmgCJj4d0VdOUk=";
-    })
-  ];
+  # Required to get the freebsd-ports patch to apply.
+  # There seem to be inconsistent line endings in this project - just converting the patch doesn't work
+  prePatch = lib.optionalString stdenv.hostPlatform.isFreeBSD ''
+    ${buildPackages.dos2unix}/bin/dos2unix src/LercLib/fpl_EsriHuffman.cpp src/LercLib/fpl_Lerc2Ext.cpp
+  '';
+
+  patches =
+    [
+      # https://github.com/Esri/lerc/pull/227
+      (fetchpatch {
+        name = "use-cmake-install-full-dir.patch";
+        url = "https://github.com/Esri/lerc/commit/5462ca7f7dfb38c65e16f5abfd96873af177a0f8.patch";
+        hash = "sha256-qaNR3QwLe0AB6vu1nXOh9KhlPdWM3DmgCJj4d0VdOUk=";
+      })
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isFreeBSD [
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/freebsd/freebsd-ports/ee9e39ceb1af729ac33854b5f3de652cb5ce0eca/graphics/lerc/files/patch-_assert";
+        hash = "sha256-agvGqgIsKS8v43UZdTVxDRDGbZdj2+AzKoQONvQumB4=";
+        extraPrefix = "";
```

Comment:
```suggestion

```

Reply:
The default is not an empty string, the default is null, which is important. This works around the fact that freebsd-ports patches aren't for patch -p1, they're for patch -p0, without having to make the entire derivation patchPhase use -p0.